doc/developers/contributing.rst

```diff
@@ -248,19 +248,25 @@ modifying code and submitting a PR:
    and start making changes. Always use a feature branch. It's good
    practice to never work on the ``master`` branch!
 
-9. Develop the feature on your feature branch on your computer, using Git to
-   do the version control. When you're done editing, add changed files using
-   ``git add`` and then ``git commit``::
+9. (**Optional**) Install `pre-commit <https://pre-commit.com/#install>`_ to
+   run code style checks before each commit::
 
-       $ git add modified_files
-       $ git commit
+        $ pip install pre-commit
+        $ pre-commit install
```

Comment:
I think you might need to mention that occasionally (particularly when creating merge commits), you might need to use `git commit -n` to disable pre-commits.

Reply:
What error message will users get the first time?
If it's explicit enough so that a simple google search give them the answer, I'd leave it out.

This section is aimed at new contributors and we need to be careful about not overloading with information (that was an issue in the previous version of the guide that we tried to trim down). Also I honestly doubt they'll ever remember "oh right I saw that in the UG"